projects/lib/src/board/sittuyinboard.cpp

```diff
@@ -301,6 +327,12 @@ Result SittuyinBoard::result()
 			return Result(Result::Draw, Side::NoSide, str);
 		}
 	}
+	if (!m_inSetUp && !canMakeNormalMove())
```

Comment:
@alwey I did not test it, but from the code it reads like a position such as `8/8/4p3/4P3/2p5/2P3p1/5k2/7K[-] w - - 0 1` would also be adjudicated as a draw, although 3.9c7 seems to explicitly restrict the adjudication to cases where only one pawn is left:
> When no other pieces but only a pawn is left on the board and it has no square to move and forming the stale-mate position, the player has the right not to take promotion and claim the draw.

So I guess something like the below should need to be added.
```suggestion
	if (!m_inSetUp && pieceCount(sideToMove(), Pawn) == 1 && !canMakeNormalMove())
```

Reply:
@ianfab : thank you for the heads up. Indeed the wording of rule 3.9c7 says "only one Pawn".
Other positions where additional Pawns are in opposition then do not match.

Reply:
I added your suggestion and also a fix for a bug in my latest patch.